src/sentry/api/endpoints/organization_events.py

```diff
@@ -154,7 +156,14 @@ def get(self, request: Request, organization) -> Response:
 
         sentry_sdk.set_tag("performance.metrics_enhanced", metrics_enhanced)
         allow_metric_aggregates = request.GET.get("preventMetricAggregates") != "1"
-        referrer = referrer if referrer in ALLOWED_EVENTS_REFERRERS else "api.organization-events"
+        # Force the referrer to "api.auth-token" for events requests authorized through a bearer token
+        referrer = (
+            API_TOKEN_REFERRER
+            if request.auth
+            else referrer
+            if referrer in ALLOWED_EVENTS_REFERRERS
+            else "api.organization-events"
+        )
```

Comment:
nit: bit hard to read once there's multiple ifs like this
```suggestion
        if request.auth:
            referrer = API_TOKEN_REFERRER
        elif referrer not in ALLOWED_EVENTS_REFERRERS:
            referrer = "api.organization-events"
```